clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```
@@ -453,6 +455,11 @@ public KafkaProducer(Properties properties, Serializer<K> keySerializer, Seriali
         } catch (Throwable t) {
             // call close methods if internal objects are already constructed this is to prevent resource leak. see KAFKA-2121
             close(Duration.ofMillis(0), true);
+            // throw RetriableException if due to no resolvable bootstrap server
+            if (t instanceof ConfigException && t.getCause() instanceof RetriableException) {
+                throw new NetworkException("Failed to construct Kafka producer due to no resolvable bootstrap server. "
```

can we re-throw the cause exception? 
throw (RetriableException) t.getCause()

---

Move this RetriableException ahead to fail fast and retry. Remove it here.